tools/ratbagctl.in

```diff
@@ -1540,21 +1540,28 @@ class RatbagParserRoot(object):
         self.parser.print_help()
 
 
+def open_ratbagd():
+    try:
+        r = Ratbagd()
+    except RatbagdDBusUnavailable:
+        print("Unable to connect to ratbagd over dbus")
+        return None
+    else:
+        if not r.devices:
+            print("No devices available.")
```

Comment:
https://docs.python.org/3/tutorial/errors.html
> The use of the `else` clause is better than adding additional code to the `try` clause because it avoids accidentally catching an exception that wasn’t raised by the code being protected by the `try ... except` statement.

I have been given example of these by the python core developer who helped me on the argparse mess. And I must say that I agree with the above statement.